cluster/kubernetes/kubernetes.go

```diff
@@ -110,6 +92,7 @@ type Cluster struct {
 
 // NewCluster returns a usable cluster.
 func NewCluster(clientset k8sclient.Interface,
+	dynamicClientset k8sclientdynamic.Interface,
```

Comment:
Sorry I didn't catch this before, but ... the number of parameters is out of hand. Wouldn't it make sense to create a structure for this?

Reply:
Yes, or accept an interface, and construct the clientset from outside (the problem there might be that the clientset interfaces have duplicate method definitions).

Reply:
> the clientset interfaces have duplicate method definitions

Ah yes, every single clientset has Discovery(), so it can't be done on that level. It might be possible to embed the more specific API-level interfaces ..

Reply:
Sounds good!

Regardless, I think it probably makes sense to transform the parameters of `NewCluster` into a struct (although maybe not in this PR)